### ISSUE TITLE
libtree-sitter-cedarscript.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by devsoft-co-za** [CEDARScript/cedarscript-integration-aider#2]
I am getting this error on python 3.12.6 - is there something I need to install? Should I just give up and wait for aider integration? 
Failed to load native library: 
 /aider_venv/lib/python3.12/site-packages/cedarscript_grammar/libtree-sitter-cedarscript.so: cannot open shared object file: No such file or directory

**Comment by elifarley** [comment]
You could try this inside the container:
```shell
pip install --upgrade --force-reinstall cedarscript-grammar
```

**Comment by devsoft-co-za** [comment]
Thanks - I tried that after installing aider using the new in a new conda environment: 
`conda create -n aider_cedarscript_env python=3.12.6`
`conda activate aider_cedarscript_env`
`pip install git+https://github.com/elifarley/aider@cedarscript aider-chat`
The command `aider --model deepseek/deepseek-chat --edit-format cedarscript` works and I see some CEDARSCRIPT being generated, but then the error: ``

I tried your suggestion:
`pip install --upgrade --force-reinstall cedarscript-grammar`
But received the following error message: 
`Collecting cedarscript-grammar
  Using cached cedarscript_grammar-0.5.0-py3-none-any.whl.metadata (23 kB)
Collecting tree-sitter~=0.21 (from cedarscript-grammar)
  Using cached tree_sitter-0.23.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.8 kB)
Using cached cedarscript_grammar-0.5.0-py3-none-any.whl (91 kB)
Using cached tree_sitter-0.23.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (571 kB)
Installing collected packages: tree-sitter, cedarscript-grammar
  Attempting uninstall: tree-sitter
    Found existing installation: tree-sitter 0.21.3
    Uninstalling tree-sitter-0.21.3:
      Successfully uninstalled tree-sitter-0.21.3
  Attempting uninstall: cedarscript-grammar
    Found existing installation: cedarscript-grammar 0.5.0
    Uninstalling cedarscript-grammar-0.5.0:
      Successfully uninstalled cedarscript-grammar-0.5.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
aider-chat 0.58.2.dev669+gae93ecdd requires tree-sitter==0.21.3, but you have tree-sitter 0.23.2 which is incompatible.
Successfully installed cedarscript-grammar-0.5.0 tree-sitter-0.23.2`

After that, running aider again produces the same error: 
`[find_cedar_commands] Script block count: 1
[cedarscript_grammar] Loading native CEDARScript parsing lib from /opt/miniconda3/envs/aider_cedarscript_env/lib/python3.12/site-packages/cedarscript_grammar/libtree-sitter-cedarscript.so
Failed to load native library: /opt/miniconda3/envs/aider_cedarscript_env/lib/python3.12/site-packages/cedarscript_grammar/libtree-sitter-cedarscript.so: cannot open shared object file: No
such file or directory`

I am going to clear my conda cache and try again (you can see it's using cached packages, maybe corrupt?) 
I did already get the same errors using venv so it's not conda specific. I will try with venv also using the latest recommended install command and report back - if none of those work I am stumped, I suspect Arch/Manjaro doing something - the libtree-sitter-cedarscript.so file is present in the environment binary directory just not being picked up. I use conda and venv all the time though and so far no issues with other packages installed via pip. AMD64 laptop. Up-to-date Manjaro installation.

**Comment by devsoft-co-za** [comment]
After clearing conda, pip cache, re-installing everything using venv and conda I still get the same error: 
`libtree-sitter-cedarscript.so: cannot open shared object file: No such file or directory` 

Any suggestions welcome.

**Comment by elifarley** [comment]
try to check the file size and output of `file <so-file>`

**Comment by devsoft-co-za** [comment]
`file /opt/miniconda3/envs/aider_cedarscript_env/lib/python3.12/site-packages/cedarscript_grammar/libtree-sitter-cedarscript.so`

`/opt/miniconda3/envs/aider_cedarscript_env/lib/python3.12/site-packages/cedarscript_grammar/libtree-sitter-cedarscript.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, BuildID[sha1]=167a842443c8a279691fd7c48f2f45956e346d4b, not stripped`

**Comment by devsoft-co-za** [comment]
OK so it seems I have the ARM64 version. How did this happen and how can I fix it? Any attempt to re-install ends up the same. Could I have done this myself by mis-configuring while compiling the source once? 

First time this happened to me. I will look at it some more. At least I know what the problem is now.

**Comment by devsoft-co-za** [comment]
OK I have tried various things - I can't get cedarscript-grammar to compile, some sort of npm problem (I don't use that much so no idea) but I did try to get pip to compile it for me using ```--no-binary=:all:```, and a few other things like pip version, npm version... No luck.  

Anyway it's a libtree-sitter-cedarscript.so problem so please close this issue if it's not relevant here. 

The only thing I want to know is, does CEDARSCRIPT with Aider actually work on X86-64? I just wonder if it's a new mac specific program and that's why I can't get it to run.

**Comment by elifarley** [comment]
I need to compile a shared lib for the `x64` architecture...
Or you could help me and compile it.
Can you please check out CEDARScript grammar (https://github.com/CEDARScript/cedarscript-grammar.git)

and then run

```shell
make build
```

This should create a shared lib file at `src/cedarscript_grammar/libtree-sitter-cedarscript.so`

If it runs into problems, can you send the logs please?

Thanks for your effort to solve this problem!

**Comment by elifarley** [comment]
[This](https://github.com/CEDARScript/cedarscript-grammar/commit/d7e82bde13d71c2746f074cee4f79ae64f4a80bd) should fix it.

Can you please try again?
This time you'll have `cedarscript-grammar 0.5.1` installed, which supports amd64

**Comment by devsoft-co-za** [comment]
Thanks so much @elifarley - when I saw this `Downloading tree_sitter-0.21.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (502 kB)` I knew it was fixed already!

Sorry I didn't do the compiling when you asked - weekends are the busiest time for me at work, so I didn't see the email until now. 

Just ran the --edit-format cedarscript and did my first commit. It works!! Now I am a 10x developer!

Thank you so much for the help getting this working. I really believe CEDARSCRIPT is the way forward getting LLM's to do code for us.
